code/modules/mob/living/simple_animal/hostile/plague_rat.dm

```diff
@@ -0,0 +1,88 @@
+/mob/living/simple_animal/hostile/plaguerat
+	name = "rat"
+	desc = "It's a nasty, ugly, evil, disease-ridden rodent with anger issues."
+	icon_state = "mouse_gray"
+	icon_living = "mouse_gray"
+	icon_dead = "mouse_gray_dead"
+	speak = list("Skree!","SKREEE!","Squeak?")
+	speak_emote = list("squeaks")
+	emote_hear = list("Hisses.")
+	emote_see = list("runs in a circle.", "stands on its hind legs.")
+	melee_damage_lower = 5 //stronk
+	melee_damage_upper = 5
+	obj_damage = 5
+	speak_chance = 1
+	turns_per_move = 5
+	see_in_dark = 6
+	maxHealth = 20
+	health = 20
+	butcher_results = list(/obj/item/reagent_containers/food/snacks/meat/slab/mouse = 1)
+	density = FALSE
+	ventcrawler = VENTCRAWLER_ALWAYS
+	pass_flags = PASSTABLE | PASSGRILLE | PASSMOB
+	mob_size = MOB_SIZE_TINY
+	mob_biotypes = list(MOB_ORGANIC,MOB_BEAST)
+	faction = list("rat")
+	var/playstyle_string = "<span class='big bold'>You are a plague rat,</span></b> \
+							You need to infect people with plague, buy getting plague cells in their bloodstream. \
+							You can do it by licking their food, or just biting them. \
+							Also you can bite dead bodies to slightly heal yourself. \
+							But remember that you are very fragile - after all, you are a small rat.</b>"
+
+
+/mob/living/simple_animal/hostile/plaguerat/AttackingTarget()
+	..()
+	var/mob/living/L = target
+	if(isliving(target))
+		if(target.reagents)
+			L.reagents.add_reagent(/datum/reagent/plaguebacteria, 3)
```

Comment:
People in protective equipment should be immune, maybe just remove this entire section and only add reagents if rats successfully lick someone.

Reply:
I don’t know how do check does someone have protective equipment, and also as i said rat transfer reagents to people not through licking them, but through biting them. Licking is only for food/reagent containers.

Reply:
I think you can do something along the lines of

```
    if(isliving(target) && target.can_inject(src, TRUE))
```

To check if they're wearing any protective equipment